bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,656 @@
+            List<Channel> channels = new ArrayList<>(this.getThing().getChannels());
+            channels.removeIf(c -> (c.getUID().getId().equals(HUMIDITY)));
+            channels.removeIf(c -> (c.getUID().getId().equals(PROGRAM_MODE)));
+            updateThing(editThing().withChannels(channels).build());
+        }
+
+        if (errorMsg == null) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, rthermData.getName() + " " + rthermData.getModel());
+            initialized = 1;
+            startAutomaticRefresh();
+            if (!(config.disableLogs == 1 && config.disableHumidity == 1)) {
+                startAutomaticLogRefresh();
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * Start the job to periodically update data from the thermostat
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                try {
+                    // Request new data from the thermostat
+                    retryCounter = 0;
+                    Object result = getRadioThermostatData(DEFAULT_RESOURCE);
+                    if (result instanceof RadioThermostatJsonResponse) {
+                        rthermData.setThermostatData((RadioThermostatJsonResponse) result);
+
+                        // Update all channels with the new data
+                        updateAllChannels();
+                    }
+                } catch (Exception e) {
+                    logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            };
+
+            RadioThermostatConfiguration config = getConfigAs(RadioThermostatConfiguration.class);
+            int delay = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+    
+    /**
+     * Start the job to periodically update humidity and runtime date from the thermostat
+     */
+    private void startAutomaticLogRefresh() {
+        if (logRefreshJob == null || logRefreshJob.isCancelled()) {
+            RadioThermostatConfiguration config = getConfigAs(RadioThermostatConfiguration.class);
+            Runnable runnable = () -> {
+                try {
+                    // Request humidity data from the thermostat if we are a CT80
+                    if (isCT80() && config.disableHumidity != 1) {   
+                        retryCounter = 0;
+                        Object result = getRadioThermostatData(HUMIDITY_RESOURCE);
+                        if (result instanceof RadioThermostatJsonHumidity) {
+                            rthermData.setHumidity(((RadioThermostatJsonHumidity) result).getHumidity());
+                        }
+                        Thread.sleep(2000);
```

Comment:
This will block one of openHAB's scheduler Threads. You should schedule another one-shot task with a 2 sec delay.